gnd/src/main/java/com/google/android/gnd/ui/recorddetails/RecordDetailsViewModel.java

```diff
@@ -52,89 +57,47 @@
   RecordDetailsViewModel(DataRepository dataRepository) {
     this.dataRepository = dataRepository;
 
-    this.recordDetailsRequest = BehaviorSubject.create();
-
-    this.recordStream =
-        recordDetailsRequest
-          .switchMap(a ->
-              this.dataRepository.getRecordDetails(a.getProjectId(), a.getFeatureId(), a.getRecordId()))
-          .toFlowable(BackpressureStrategy.LATEST);
-
-    //TODO: Refactor to expose the fetched record directly.
-    this.records = LiveDataReactiveStreams.fromPublisher(
-        recordStream
-    );
-
-    this.progressBarVisibility = LiveDataReactiveStreams.fromPublisher(
-        recordStream.map(this::toProgressBarVisibility)
-        .onErrorReturn(x -> View.GONE)
-    );
-
-    this.toolbarTitle = LiveDataReactiveStreams.fromPublisher(
-        recordStream.map(this::toToolbarTitle)
-        .onErrorReturn(x -> "")
-    );
-
-    this.toolbarSubtitle = LiveDataReactiveStreams.fromPublisher(
-        recordStream.map(this::toToolbarSubtitle)
-        .onErrorReturn(x -> "")
-    );
-
-    this.formNameView = LiveDataReactiveStreams.fromPublisher(
-        recordStream.map(this::toFormNameView)
-        .onErrorReturn(x -> "")
-    );
-
-    recordDetailsRequest.subscribe(args -> {
-      Log.d(TAG, "" + args);
-    });
+    this.recordProcessor = BehaviorProcessor.create();
+
+    // TODO: Refactor to expose the fetched record directly.
+    this.records = LiveDataReactiveStreams.fromPublisher(recordProcessor);
+
+    this.progressBarVisibility =
+        LiveDataReactiveStreams.fromPublisher(
+            recordProcessor.map(this::toProgressBarVisibility).onErrorReturn(x -> View.GONE));
+
+    this.toolbarTitle =
+        LiveDataReactiveStreams.fromPublisher(
+            recordProcessor.map(this::toToolbarTitle).onErrorReturn(x -> ""));
+
+    this.toolbarSubtitle =
+        LiveDataReactiveStreams.fromPublisher(
+            recordProcessor.map(this::toToolbarSubtitle).onErrorReturn(x -> ""));
+
+    this.formNameView =
+        LiveDataReactiveStreams.fromPublisher(
+            recordProcessor.map(this::toFormNameView).onErrorReturn(x -> ""));
   }
 
-  public void getRecordDetailsRequest(RecordDetailsFragmentArgs args) {
-    this.recordDetailsRequest.onNext(args);
+  public void loadRecordDetails(RecordDetailsFragmentArgs args) {
+    this.dataRepository
+        .getRecordDetails(args.getProjectId(), args.getFeatureId(), args.getRecordId())
+        .subscribe(record -> this.recordProcessor.onNext(record));
```

Comment:
Any time we subscribe we also need to keep track of the subscription and dispose of it at the proper point. Doing this would actually put us back where we started, with subscriptions being created each time loadRecordDetails() is called. You probably get an Android Studio warning here that we're not doing anything with the return value of subscribe().

Why did you change this back to subscribe() from the switchMapped solution?

Reply:
Ahh, I see, duh. I got rid of it just because it allowed me to drop the `RecordDetailsRequest` and `RecordStream` properties, which I felt simplified the class since they're really just temporary conversions/type negotiations. Obviously not worthwhile if it makes the solution incorrect 😵

Reply:
Yeah, the reason we added them was to not need to manage Disposable subscriptions here.

Reply:
Sounds good, I've switched back to the old, new approach.